src/metabase/upload.clj

```diff
@@ -252,10 +240,18 @@
                    (= (normalize-column-name (:name field)) auto-pk-column-name))
                  (t2/select :model/Field :table_id table-id :active true))))
 
+(defn- type-relaxer
+  "Given a map of {value-type -> predicate}, return a reducing fn which updates our inferred schema using the next row."
+  [settings]
+  (let [relax (partial relax-type (settings->type->check settings))]
```

Comment:
I see that I missed the chance to complain about it when it was introduced, but two arrows in the one name is doing my head in. I see that it's `(settings) -> (type->check)`, but I keep wondering if it should be parsed the other way

Reply:
Haskell has convinced me that this ambiguity doesn't matter - arrows are associative 😄